src/Entity/Repository/MeshDescriptorRepository.php

```diff
@@ -773,4 +775,135 @@ public function getIliosMeshDescriptorsById(array $ids): array
             return $descriptor;
         }, $fullDescriptors);
     }
+
+    /**
+     * @return array
+     * @throws DBALException
+     */
+    public function exportMeshDescriptors(): array
+    {
+        $sql = <<<EOL
```

Comment:
Is this syntax needed here? Seems equivelent to 
```
$qb = $this->_em->createQueryBuilder();
$qb->select('m.id, m.name, m.annotation, m.createdAt, m.updatedAt, m.deleted')
  ->from(MeshDescriptor::class, 'm');
  return $qb->getQuery()->getScalarResult();
```

Reply:
_almost_ identical. the query builder does some light output massaging, most notably converting timstamps to PHP `DateTime` objects. which we absolutely don't want  here, since we'd have to convert them to string representations ourselves.

Reply:
Oh, I swear when I tested it with `getScalarResult` they were strings and not DateTime. It was late though, carry on! I accept this HEREDOC madness.

Reply:
alright, will give it another go, too...

Reply:
oh wild, this works. i started with querybuilder too, must have invoked a different fetch* method that wasn't compatible. converting all of them to the proposed template now.